app/models/ability.rb

```diff
@@ -0,0 +1,39 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    include CanCan::Ability
+    return unless user.present?
+
+    can :read, :all
+    can :create, Comment
+    can :create, Post
+    can :destroy, Post do |post|
+      post.try(:user) == user
+    end
+    can :destroy, Comment do |comment|
+      comment.try(:user) == user
+    end
+    return unless user.admin?
+
+    can :manage, :all
```

Comment:
- [ ] Hello @stephnna, great job so far 👏🏻 , The authorization has been implemented fine using can can can gem but I have not been able to check it on my computer due to the error I am getting while creating the base of data, please correct this:

![Screenshot from 2022-12-07 17-55-11](https://user-images.githubusercontent.com/79058364/206229473-41ff15b1-d24c-4210-9ccc-cb0bb1b931ce.png)


- Note: If it works fine on your end, don't worry, attach a screenshot for the next reviewer to approve 😉 👍🏻

Reply:
I have fixe the error, it works fine now
Here is the screen short
![Code](https://user-images.githubusercontent.com/96915150/206410253-96a0645f-7ee0-4e24-b160-dedd0c168697.PNG)